app.py

```diff
@@ -19,6 +25,20 @@ async def root():
 async def get_name(name):
     return JSONResponse({"name": name}, status_code=200)
 
+@app.post("/operation")
+async def perform_operation_endpoint(request: OperationRequest):
+    result = perform_operation(request.a, request.b, request.operator)
+    if "Error" in str(result):
+        raise HTTPException(status_code=400, detail=result)
+    return JSONResponse({"result": result})
+
+@app.get("/operation/{a}/{b}/{operator}")
+async def perform_operation_get(a: int, b: int, operator: str):
+    result = perform_operation(a, b, operator)
+    if "Error" in str(result):
+        raise HTTPException(status_code=400, detail=result)
+    return JSONResponse({"result": result})
+
```

Comment:
You can remove the second one as the first @app.post("/operation") meets our requirement.